cmd/fs-v1-multipart.go

```diff
@@ -19,16 +19,61 @@ package cmd
 import (
 	"crypto/md5"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"hash"
 	"io"
-	"path"
+	"io/ioutil"
+	"os"
+	pathutil "path"
 	"strings"
 	"time"
 
 	"github.com/minio/sha256-simd"
 )
 
+// listMultipartUploadIDs - list all the upload ids from a marker up to 'count'.
+func (fs fsObjects) listMultipartUploadIDs(bucketName, objectName, uploadIDMarker string, count int) ([]uploadMetadata, bool, error) {
+	var uploads []uploadMetadata
+
```

Comment:
But we are holding a lock on the parent wouldn't that be sufficient? why would a EX succeed with in the same process if the parent is locked.. ? EX on `uploads.json` gets only modified when parent is granted..